File: planner/app/src/main/java/com/sunilson/firenote/views/CategoryVisibilityView.java
package com.sunilson.firenote.views;

import android.content.Context;
import android.graphics.Paint;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.CheckBox;
import android.widget.LinearLayout;
import android.widget.Checkable;
import android.widget.TextView;

import com.sunilson.firenote.R;

/**
 * @author Linus Weiss
 */

/**
 * View used for elements in a list of categories. Implements Checkable, so it can be easily selected
 */
public class CategoryVisibilityView extends LinearLayout implements Checkable {

    private View v;
    private TextView textView;
    private CheckBox checkBox;
    private boolean checked = false;

    public CategoryVisibilityView(Context context) {
        super(context);

        LayoutInflater inflater = LayoutInflater.from(context);
        v = inflater.inflate(R.layout.category_list_layout, this, true);
        textView = (TextView) v.findViewById(R.id.category_element_text);
        checkBox = (CheckBox) v.findViewById(R.id.category_element_checkBox);
    }

    /**
     * Set element to given checked state
     * @param checked True or false
     */
    @Override
    public void setChecked(boolean checked) {
        this.checked = checked;

        if (isChecked()) {
            textView.setPaintFlags(textView.getPaintFlags() | Paint.STRIKE_THRU_TEXT_FLAG);
            textView.setTextColor(getResources().getColor(R.color.text_crossed_out));
            checkBox.setChecked(false);
        } else {
            textView.setPaintFlags(textView.getPaintFlags() & ~(Paint.STRIKE_THRU_TEXT_FLAG));
            textView.setTextColor(getResources().getColor(R.color.primary_text_color));
            checkBox.setChecked(true);
        }
    }

    @Override
    public boolean isChecked() {
        return checked;
    }

    @Override
    public void toggle() {
        checked = !checked;
        setChecked(checked);
    }
}

File: planner/app/src/main/java/com/sunilson/firenote/presentation/visibilityDialog/adapters/CategoryVisibilityView.kt
package com.sunilson.firenote.presentation.visibilityDialog.adapters

import android.content.Context
import android.graphics.Paint
import android.view.LayoutInflater
import android.view.View
import android.widget.Checkable
import android.widget.LinearLayout
import com.sunilson.firenote.R
import kotlinx.android.synthetic.main.category_list_layout.view.*

class CategoryVisibilityView(context: Context) : LinearLayout(context), Checkable {

    val v: View
    private var mChecked: Boolean = false

    init {
        v = LayoutInflater.from(context).inflate(R.layout.category_list_layout, this, true)
    }

    override fun isChecked(): Boolean = mChecked

    override fun toggle() {
        mChecked != mChecked
        this.isChecked = mChecked
    }

    override fun setChecked(checked: Boolean) {
        mChecked = checked

        if (mChecked) {
            category_element_text.paintFlags = category_element_text.paintFlags or Paint.STRIKE_THRU_TEXT_FLAG
            category_element_text.setTextColor(resources.getColor(R.color.text_crossed_out))
            category_element_checkBox.isChecked = false
        } else {
            category_element_text.paintFlags = category_element_text.paintFlags and  Paint.STRIKE_THRU_TEXT_FLAG.inv()
            category_element_text.setTextColor(resources.getColor(R.color.primary_text_color))
            category_element_checkBox.isChecked = true
        }
    }
}